zrcs/zrc-7.md

```diff
@@ -0,0 +1,183 @@
+
+## IV. Specification
+
+The metadata specification describes:
+
+1. The metadata standard that should be found when navigating to a particular non fungible token's token_uri.
+2. The localization interface for developers to implement and ecosystems to consume.
+3. The dynamic string literals that may be present in the URL string, such as {id} and {locale}
+
+### A. Rationale
+
+Compared to ERC-1155, the JSON Metadata schema allows to indicate the MIME type of the files pointed by any URI field. This is to allow clients to display appropriately the resource without having to first query it to find out the MIME type.
+
+Metadata localization should be standardized to increase presentation uniformity across all languages. As such, a simple overlay method is proposed to enable localization.
+
+NFT's are broken into two parts. The first is the blockchain smart contract, which contains the association between users and a token. Each token when examained points off-chain to an external resource which contains the metadata for that particular token. This could be on the surface web or using a decentralised storage solution such as IPFS or Arweave.
+
+ZRC-6 gives the developer the ability to set an ```base_uri``` which exposes a field for the developer to insert a URI they control. When navigating to ```base_uri/{id}``` for a given ```token_id```, the response returned should be a standardised JSON structure for it to be considered ZRC-7 compliant.
+
+The main advantages of keeping metadata offchain are as follows.
+
+* Mutability
+  * Owner can update the metadata at anytime without a chain call (changing hashes of IPFS and Arweave resources)
+* Storage
+  * Owner can store more data offchain than onchain and for a cheaper cost to maintain and amend.
+
+The main disadvantages of keeping metadata offchain are as follows.
+
+* Mutability
+  * Owner can maliciously update the metadata at anytime without a chain call.
+
+### B. Metadata Structure Example
+
+ZRC-6 has a field whereby developers can store a resource where all of the base metadata can be found. This can either be an API or direct storage. The ```token_uri``` returned from the contract for a particular ```token_id``` is ```base_uri/token_id``` appended together. 
+
+If ```base_uri``` is https://creatures-api.zilliqa.com/api/creature/ and ```token_id``` is 1, then `token_uri` for token id 1 becomes ```https://creatures-api.zilliqa.com/api/creature/1```.
+
+```json
+{
+    "name": "Advertising Space",
+    "description": "Each token represents a unique advertising space in the city.",
+    "resource": "ipfs://QmSjJGa7zXUbixvYJpgkRkaSCYEBtSwgVtfzkoD3QkSsty",
+    "resource_mimetype": "image/png",
+    "external_url": "https://example.com",
+    "external_description": "A generic headline from the project shared between all tokens.",
```

Comment:
`external_description` might be unnecessary.
Can you explain why this is necessary?

Reply:
```description``` describes a particular asset as a 1:1 relation , whereas ```external_description``` is a 1 to many relationship where many tokens share the same description. It is necessary because it provides more infomation that isn't covered by ERC-1155. Currently marketplaces allow contract owners to edit their description via the website, but it's nicer if it's covered by the metadata. The disadvantages of this is the duplication over N amount of metadata files.